TodoApp/src/TodoApp.Contract/Repositories/IItemRepository.cs

```diff
@@ -1,15 +1,16 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
 using TodoApp.Contract.Models;
 
 namespace TodoApp.Contract.Repositories
 {
     public interface IItemRepository
     {
-        IEnumerable<Item> GetAll();
-        Item Get(Guid id);
-        Item Update(Guid id, Item item);
-        Item Add(Item item);
+        Task<Item[]> GetAll();
```

Comment:
I liked `IEnumerable<Item>` more, why have you changed the type?

Reply:
Well, I guess I had some problems with type conversions when there was IEnumerable, but I' not really sure now... 😟 Actually, in task 3, I converted it to List 😄 do you think IEnumerable is still better than List? Or that List is ok? 😇

Reply:
As always, it depends. There are several aspects mostly connected with the content of the collection and usage. But in general:
* you want to avoid [premature materialization](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/intermediate-materialization)
* you want to control what your interface users do with the data (returning list gives an easy way to `Add` some entities (after the `GetAll` method is called) and indexed access) and returning general interfaces makes it harder to unintentionally spoil a collection

Let's discuss this in person as well 😃